xdem/coreg.py

```diff
@@ -596,6 +593,99 @@ def to_matrix(self) -> np.ndarray:
         """Convert the transform to a 4x4 transformation matrix."""
         return self._to_matrix_func()
 
+    def centroid(self) -> Optional[tuple[float, float, float]]:
+        """Get the centroid of the coregistration, if defined."""
+        meta_centroid = self._meta.get("centroid")
+
+        if meta_centroid is None:
+            return None
+
+        # Unpack the centroid in case it is in an unexpected format (an array, list or something else).
+        return (meta_centroid[0], meta_centroid[1], meta_centroid[2])
+
+    def residuals(self, reference_dem: Union[np.ndarray, np.ma.masked_array],
+                  dem_to_be_aligned: Union[np.ndarray, np.ma.masked_array],
+                  inlier_mask: Optional[np.ndarray] = None,
+                  transform: Optional[rio.transform.Affine] = None) -> np.ndarray:
+        """
+        Calculate the residual offsets (the difference) between two DEMs after applying the transformation.
+
+        :param reference_dem: 2D array of elevation values acting reference. 
+        :param dem_to_be_aligned: 2D array of elevation values to be aligned.
+        :param inlier_mask: Optional. 2D boolean array of areas to include in the analysis (inliers=True).
+        :param transform: Optional. Transform of the reference_dem. Mandatory in some cases.
+
+        :returns: A 1D array of finite residuals.
+        """
+        # Use the transform to correct the DEM to be aligned.
+        aligned_dem = self.apply(dem_to_be_aligned, transform=transform)
+
+        # Format the reference DEM
+        ref_arr, ref_mask = xdem.spatial_tools.get_array_and_mask(reference_dem)
+
+        if inlier_mask is None:
+            inlier_mask = np.ones(ref_arr.shape, dtype=bool)
+
+        # Create the full inlier mask (manual inliers plus non-nans)
+        full_mask = (~ref_mask) & np.isfinite(aligned_dem) & inlier_mask
+
+        # Calculate the DEM difference
+        diff = ref_arr - aligned_dem
+
+        # Return the difference values within the full inlier mask
+        return diff[full_mask]
+
+    def error(self, reference_dem: Union[np.ndarray, np.ma.masked_array],
+              dem_to_be_aligned: Union[np.ndarray, np.ma.masked_array],
+              error_type: str = "nmad",
```

Comment:
Looks really great! Just one comment is that sometimes one might want more than just one statistics, and it is counter-productive to calculate the residuals each time. So it would be nice to be able to request several statistics at once rather than only one...